### DISCUSSION TITLE
Open nvim on VimEnter

### DISCUSSION THREAD

**Discussion by reyniersbram** [nvim-tree/nvim-tree.lua#2215] [Q&A]
I added [this](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#open-for-directories-and-change-neovims-directory) code (the current window variant) to my nvim-tree config to open nvim-tree when I open a directory in NeoVim, but it doesn't seem to do anything. Yes, I also put this line below:
```lua
vim.api.nvim_create_autocmd({ "VimEnter" }, { callback = open_nvim_tree })
```
but when I comment it out, I get the exact same behaviour.

I tinkered with the if-test a little bit, but I'm unable to grow any wiser from it.

Can someone explain to me what exactly is happening in this snippet of code?

Thanks in advance

**Comment by alex-courtis** [comment]
Some avenues of investigation:

Add a `print` statement inside your open_nvim_tree function to ensure it's actually running.

Ensure you're not using any lazy loading plugin startup that may interfere with your autocommands.

Ensure netrw is disabled, see `:help nvim-tree-quickstart`

Disable directory hijacking which may interefere with your startup, see `:help nvim-tree.hijack_directories`
